### PR TITLE
Add Email module

### PR DESCRIPTION
**Pull request by zaucker**
- Build multi-part message.

### CODE REVIEW

File: lib/Agrammon/Email.pm6 [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+    has $!to;
+    has $!from;
+    has $!attachment;
+#    has $!filename;
+    has $!mail;
+
+    submethod TWEAK( :$attachment, :$msg, :$filename, :$to, :$from, :$subject --> Nil) {
+        $!to = $to;
+        $!from = $from;
+        $!subject = $subject;
```
Comment (jnthn) [inline lib/Agrammon/Email.pm6:18]: Can do `:$!to, :$!from, :$!subject` in the signature and save these three lines.

File: lib/Agrammon/Email.pm6 [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+                :to([ $!to ]),
+                :message(~$!mail),
+            );
+
+            .quit;
+
+            CATCH {
+                when X::Net::SMTP::Client::Async {
+                    note "Unable to send email message: $_";
+                    .quit
```
Comment (jnthn) [inline lib/Agrammon/Email.pm6:70]: Since `CATCH` binds the current exception to `$_`, this will call `.quit` on the wrong thing.

Comment (jnthn) [reply lib/Agrammon/Email.pm6:70]: Maybe better is above to change it to `LEAVE .quit`, and then you don't need it here at all.

File: lib/Agrammon/Email.pm6 (hunk 1 above)
Comment (zaucker) [reply lib/Agrammon/Email.pm6:18]: Done.

File: lib/Agrammon/Email.pm6 (hunk 2 above)
Comment (zaucker) [reply lib/Agrammon/Email.pm6]: Done. So this is a bug in the README of https://modules.raku.org/dist/Net::SMTP::Client::Async:cpan:HANENKAMP ?

Comment (jnthn) [reply lib/Agrammon/Email.pm6]: Yes.

File: lib/Agrammon/Email.pm6 [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+
+class Agrammon::Email {
+    has $!msg;
+    has $!subject;
+    has $!to;
+    has $!from;
+    has $!attachment;
+    has $!mail;
+
+    submethod TWEAK( :$attachment, :$msg, :$filename, :$!to, :$!from, :$!subject --> Nil) {
```
Comment (jnthn) [inline lib/Agrammon/Email.pm6:14]: Hm, looking at this again, I wonder if some of these should be marked required to ensure correct usage (as nameds are all optional by default, but it looks like only the attachment is really optional).

Comment (zaucker) [reply lib/Agrammon/Email.pm6]: Fixed. Gave the filename a default.

**Review by jnthn — APPROVED** [review]

**Pull request merged**